source/tests/system/nixnet_can_driver_api_tests.cpp

```diff
@@ -0,0 +1,145 @@
+#include <gmock/gmock.h>
+#include <google/protobuf/util/time_util.h>
+#include <gtest/gtest.h>
+#undef interface
+#include <nixnet/nixnet_client.h>
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "device_server.h"
+#include "enumerate_devices.h"
+#include "nixnet_utilities.h"
+
+using namespace nixnet_grpc;
+namespace client = nixnet_grpc::experimental::client;
+namespace pb = google::protobuf;
+using namespace ::testing;
+using namespace ::nixnet_utilities;
+
+namespace ni {
+namespace tests {
+namespace system {
+namespace {
+
+using u8 = pb::uint8;
+using u32 = pb::uint32;
+using u64 = pb::uint64;
+using f64 = double;
+
+class NiXnetCANDriverApiTests : public ::testing::Test {
+ protected:
+  NiXnetCANDriverApiTests()
+      : device_server_(DeviceServerInterface::Singleton()),
+        stub_(NiXnet::NewStub(device_server_->InProcessChannel()))
+  {
+    device_server_->ResetServer();
+  }
+  virtual ~NiXnetCANDriverApiTests() {}
+
+  void TearDown() override
+  {
+    device_server_->ResetServer();
+  }
+
+  template <typename TService>
+  std::unique_ptr<typename TService::Stub> create_stub()
+  {
+    return TService::NewStub(device_server_->InProcessChannel());
+  }
+
+  std::unique_ptr<NiXnet::Stub>& stub()
+  {
+    return stub_;
+  }
+
+  void assert_can_frames_are_equal(nixnet_grpc::FrameRequest* frame1, nixnet_grpc::FrameResponse* frame2)
+  {
+    int frame1_flags = calculate_bitwise_or_of_flags(frame1->flags());
+    int frame2_flags = calculate_bitwise_or_of_flags(frame2->flags());
+    EXPECT_EQ(frame1_flags, frame2_flags);
+
+    EXPECT_EQ(frame1->type(), frame2->type());
+    EXPECT_EQ(frame1->identifier(), frame2->identifier());
+    EXPECT_EQ(frame1->payload(), frame2->payload());
+  }
+
+  void set_frame_data(nixnet_grpc::FrameRequest* frame, u64 timestamp, nixnet_grpc::FrameFlags frame_flag, u32 identifier, nixnet_grpc::FrameType frame_type, const char* payload)
+  {
+    frame->set_timestamp(timestamp);
+    frame->add_flags(frame_flag);
+    frame->set_identifier(identifier);
+    frame->set_type(frame_type);
+    frame->set_payload(payload);
+  }
+
+ private:
+  DeviceServerInterface* device_server_;
+  std::unique_ptr<NiXnet::Stub> stub_;
+};
+
+TEST_F(NiXnetCANDriverApiTests, TwoFrameExample_ConvertFramesToSignalAndBackToFramesSignlePoint_InputAndOutputFramesAreEquivalent)
```

Comment:
@asumit Looks like you misspelled "Single" in FramesSignlePoint"

Reply:
Thanks for catching it Mose, I have created PR #649 to correct it.